client-spark/spark3/src/main/java/org/apache/spark/shuffle/RssShuffleManager.java

```diff
@@ -317,6 +350,8 @@ public <K, V, C> ShuffleHandle registerShuffle(int shuffleId, ShuffleDependency<
     }
     startHeartbeat();
 
+    shuffleIdToPartitionNum.putIfAbsent(shuffleId, dependency.partitioner().numPartitions());
+    shuffleIdToNumMapTasks.putIfAbsent(shuffleId, dependency.rdd().partitions().length);
```

Comment:
`rdd.partitions.length = NumMapTasks`?

Reply:
> yes. You could refer how spark generates its numMaps in scheduler?

If we use limit operation, the part of partitions of rdd will only be executed. Is it right?